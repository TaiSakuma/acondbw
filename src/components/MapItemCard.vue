<template>
  <div class="mapitemcard">
    <div v-if="$apollo.queries.map.loading">loading...</div>
    <div v-else-if="error">Error: cannot load data</div>
    <v-card v-else-if="map" outlined hover style="max-width: 980px;">
      <div @click="$emit('expand')" style="cursor: default;">
        <v-container fluid class="pa-0">
          <v-row class="ma-0 px-0">
            <v-col cols="12" md="4" class="pa-2">
              <div class="caption grey--text">Name</div>
              <div class="font-weight-medium primary--text">
                <span @click.stop>
                  <router-link :to="'/maps/item/' + map.name" v-text="map.name"></router-link>
                </span>
              </div>
            </v-col>
            <v-col cols="6" md="4" class="pa-2">
              <div class="caption grey--text">Date posted</div>
              <div v-text="map.datePosted"></div>
            </v-col>
            <v-col cols="5" md="3" class="pa-2">
              <div class="caption grey--text">Mapper</div>
              <div v-text="map.mapper"></div>
            </v-col>
            <v-col cols="1" md="1" align-self="end" class="pa-2">
              <div v-if="collapsible">
                <v-row align="start" justify="end" class="px-1">
                  <v-tooltip bottom open-delay="800">
                    <template v-slot:activator="{ on }">
                      <v-btn
                        icon
                        @click.stop="
                      collapsed ? $emit('expand') : $emit('collapse')
                    "
                        v-on="on"
                      >
                        <v-icon>
                          {{
                          collapsed ? "mdi-chevron-down" : "mdi-chevron-up"
                          }}
                        </v-icon>
                      </v-btn>
                    </template>
                    <span>{{ collapsed ? "Expand" : "Collapse" }}</span>
                  </v-tooltip>
                </v-row>
              </div>
            </v-col>
          </v-row>
          <v-expand-transition>
            <v-row class="mx-0 mb-3 px-0 collapsible" v-show="!(collapsible && collapsed)">
              <v-col cols="12" md="8" offset-md="4" class="pa-2">
                <div class="caption grey--text">Paths</div>
                <ul v-if="map.mapFilePaths">
                  <li
                    v-for="(edgep, index) in map.mapFilePaths.edges"
                    :key="index"
                    v-text="edgep.node.path"
                  ></li>
                </ul>
              </v-col>
              <v-col cols="12" md="8" offset-md="4" class="pa-2">
                <div class="caption grey--text">Note</div>
                <div>
                  <ul v-if="map.note">
                    <li
                      v-for="(line, index) in map.note.split('\n')"
                      :key="index"
                    >{{ line.replace(/^- */, "") }}</li>
                  </ul>
                </div>
              </v-col>
              <v-col cols="12" md="8" offset-md="4" class="pa-2">
                <div class="caption grey--text">Beams</div>
                <ul v-if="map.beams">
                  <li v-for="(edgep, index) in map.beams.edges" :key="index">
                    <router-link :to="'/beams/item/' + edgep.node.name" v-text="edgep.node.name"></router-link>
                  </li>
                </ul>
              </v-col>
            </v-row>
          </v-expand-transition>
        </v-container>
      </div>
    </v-card>
    <div v-else>Nothing to show here.</div>
  </div>
</template>

<script>
import MAP from "@/graphql/Map.gql";

export default {
  name: "MapItemCard",
  props: {
    mapName: { default: null },
    collapsed: { default: false },
    collapsible: { default: false }
  },
  data() {
    return {
      map: null,
      error: null
    };
  },
  apollo: {
    map: {
      query: MAP,
      variables() {
        return {
          name: this.mapName
        };
      },
      result(result) {
        this.error = null;
        if (result.error) {
          this.error = true;
        }
      }
    }
  }
};
</script>
