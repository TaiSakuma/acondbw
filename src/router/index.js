import Vue from "vue";
import VueRouter from "vue-router";
import Home from "@/views/Home.vue";
import ProductTop from "@/views/ProductTop.vue";

import ALL_MAPS from "@/graphql/AllMaps.gql";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "home",
    component: Home,
  },
  {
    path: "/about",
    name: "about",
    component: () =>
      import(/* webpackChunkName: "about" */ "@/views/About.vue"),
  },
  {
    path: "/simulations",
    component: ProductTop,
    props: {
      title: "Simulations",
      icon: "mdi-creation",
      itemPageName: "SimulationItem",
    },
    children: [
      {
        path: "",
        name: "SimulationList",
        component: () =>
          import(
            /* webpackChunkName: "simulations" */ "@/views/SimulationList.vue"
          ),
      },
      {
        path: "item/:name",
        name: "SimulationItem",
        component: () =>
          import(
            /* webpackChunkName: "simulations" */ "@/views/SimulationItem.vue"
          ),
      },
    ],
  },
  {
    path: "/maps",
    component: ProductTop,
    props: { title: "Maps", icon: "map", itemPageName: "MapItem" },
    children: [
      {
        path: "",
        name: "MapList",
        component: () =>
          import(/* webpackChunkName: "maps" */ "@/views/MapList.vue"),
        props: {
          productTypeName: { singular: "map", plural: "maps" },
          query: ALL_MAPS,
          queryName: "allMaps",
          productIdFieldName: "mapId",
          productItemCard: "MapItemCard",
          productAddForm: "MapAddForm",
          disableAdd: process.env.VUE_APP_ACONDBW_MAP_MUTATION_DIALOG != "true",
        },
      },
      {
        path: "item/:name",
        name: "MapItem",
        component: () =>
          import(/* webpackChunkName: "maps" */ "@/views/MapItem.vue"),
      },
    ],
  },
  {
    path: "/beams",
    component: ProductTop,
    props: {
      title: "Beams",
      icon: "mdi-spotlight-beam",
      itemPageName: "BeamItem",
    },
    children: [
      {
        path: "",
        name: "BeamList",
        component: () =>
          import(/* webpackChunkName: "beams" */ "@/views/BeamList.vue"),
      },
      {
        path: "item/:name",
        name: "BeamItem",
        component: () =>
          import(/* webpackChunkName: "beams" */ "@/views/BeamItem.vue"),
      },
    ],
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes,
});

export default router;
