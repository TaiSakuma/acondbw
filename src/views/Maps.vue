<template>
  <div class="maps">
    <v-container class>
      <h2>Maps</h2>
      <div class="d-flex justify-end ma-2">
        <v-tooltip bottom open-delay="800">
          <template v-slot:activator="{ on }">
            <v-btn icon @click="showsAny = !showsAny" v-on="on">
              <v-icon>{{ showsAny ? 'mdi-unfold-less-horizontal' : 'mdi-unfold-more-horizontal' }}</v-icon>
            </v-btn>
          </template>
          <span>{{ showsAny ? 'Collapse all' : 'Expand all' }}</span>
        </v-tooltip>
      </div>
      <v-card outlined hover v-for="edge in edges" :key="edge.node.id">
        <div @click="shows[edge.node.id] = true" style="cursor: default;">
          <v-layout row wrap class="ma-0 px-3">
            <v-flex xs12 md4>
              <div class="caption grey--text">Name</div>
              <div class="font-weight-medium primary--text">
                <span @click.stop>
                  <router-link :to="'/maps/item/' + edge.node.name" v-text="edge.node.name"></router-link>
                </span>
              </div>
            </v-flex>
            <v-flex xs6 md4>
              <div class="caption grey--text">Date posted</div>
              <div v-text="edge.node.datePosted"></div>
            </v-flex>
            <v-flex xs5 md3>
              <div class="caption grey--text">Mapper</div>
              <div v-text="edge.node.mapper"></div>
            </v-flex>
            <v-flex xs1 md1 align-self-end>
              <v-btn icon @click.stop="shows[edge.node.id] = !shows[edge.node.id]">
                <v-icon>{{ shows[edge.node.id] ? 'mdi-chevron-up' : 'mdi-chevron-down' }}</v-icon>
              </v-btn>
            </v-flex>
          </v-layout>
          <v-expand-transition>
            <v-layout row wrap class="mx-0 mb-3 px-3" v-show="shows[edge.node.id]">
              <v-flex xs12 md8 offset-md-4>
                <div class="caption grey--text">Paths</div>
                <ul>
                  <li
                    v-for="(edgep, index) in edge.node.mapFilePaths.edges"
                    :key="index"
                    v-text="edgep.node.path"
                  ></li>
                </ul>
              </v-flex>
              <v-flex xs12 md8 offset-md-4>
                <div class="caption grey--text">Note</div>
                <div>
                  <ul>
                    <li
                      v-for="(line, index) in edge.node.note.split('\n')"
                      :key="index"
                    >{{ line.replace(/^- */, "") }}</li>
                  </ul>
                </div>
              </v-flex>
              <v-flex xs12 md8 offset-md-4>
                <div class="caption grey--text">Beams</div>
                <ul>
                  <li
                    v-for="(edgep, index) in edge.node.beams.edges"
                    :key="index"
                  >
                    <router-link :to="'/beams/item/' + edgep.node.name" v-text="edgep.node.name"></router-link>
                  </li>
                </ul>
              </v-flex>
            </v-layout>
          </v-expand-transition>
        </div>
      </v-card>
    </v-container>
  </div>
</template>

<script>
import axios from "axios";

export default {
  name: "maps",
  data() {
    return {
      edges: [],
      shows: {}
    };
  },
  computed: {
    showsAny: {
      get: function() {
        return Object.keys(this.shows).some(i => this.shows[i]);
      },
      set: function(v) {
        for (const k in this.shows) {
          this.shows[k] = v;
        }
      }
    }
  },
  created: function() {
    this.loadData();
  },
  methods: {
    loadData() {
      const url = process.env.VUE_APP_ACONDBS_URL;
      const query = `
        { allMaps(sort: DATE_POSTED_DESC) {
          edges {
            node {
              id
              name
              datePosted
              mapper
              note
              mapFilePaths {
                edges {
                  node {
                    path
                    note
                  }
                }
              }
              beams {
                edges {
                  node {
                    name
                  }
                }
              }
            }
          }
        }}
      `;
      axios({
        url: url,
        method: "POST",
        data: {
          query: query
        }
      }).then(response => {
        this.edges = response.data.data.allMaps.edges;
        this.shows = this.edges.reduce(
          (obj, x) => ({ ...obj, [x.node.id]: false }),
          {}
        );
      });
    }
  }
};
</script>
