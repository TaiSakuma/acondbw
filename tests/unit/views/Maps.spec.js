import Vue from "vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import { mount, createLocalVue } from "@vue/test-utils";

import Maps from "@/views/Maps.vue";
import MapItemCard from "@/components/MapItemCard.vue";

import router from "@/router";

Vue.use(Vuetify);
Vue.use(VueRouter);

describe("Maps.vue", () => {
  let localVue;

  function createWrapper() {
    return mount(Maps, {
      localVue,
      router,
      mocks: {},
      stubs: {
        MapItemCard: true,
        MapSubmitFormDialog: true
      }
    });
  }

  const allMaps = {
    edges: [
      {
        node: {
          id: "TWFwOjEwMTM=",
          name: "lat20200201"
        }
      },
      {
        node: {
          id: "TWFwOjEwMTI=",
          name: "lat20200120"
        }
      },
      {
        node: {
          id: "TWFwOjEwMDE=",
          name: "lat20190213"
        }
      }
    ]
  };

  beforeEach(function() {
    localVue = createLocalVue();
  });

  it("match snapshot", async () => {
    const wrapper = createWrapper();

    wrapper.setData({
      allMaps: allMaps,
      isCardCollapsed: {
        "TWFwOjEwMDE=": true,
        "TWFwOjEwMTI=": true,
        "TWFwOjEwMTM=": true
      }
    });
    await Vue.nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
