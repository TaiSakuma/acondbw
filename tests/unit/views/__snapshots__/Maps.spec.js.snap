// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps.vue match snapshot 1`] = `
<div class="maps">
  <div class="container container--fluid">
    <h2>Maps</h2>
    <mapsubmitformdialog-stub></mapsubmitformdialog-stub>
    <div class="d-flex justify-end ma-2" style="max-width: 980px;"><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-unfold-more-horizontal theme--light"></i></span></button><span class="v-tooltip v-tooltip--bottom"><!----></span></div>
    <div>
      <div>
        <mapitemcard-stub mapname="lat20200201" collapsed="true" collapsible="true"></mapitemcard-stub>
        <mapitemcard-stub mapname="lat20200120" collapsed="true" collapsible="true"></mapitemcard-stub>
        <mapitemcard-stub mapname="lat20190213" collapsed="true" collapsible="true"></mapitemcard-stub>
      </div>
    </div>
  </div>
</div>
`;
